gulpfile.ts

```diff
@@ -36,52 +45,108 @@ async function fetchContentsByPath(repoPath = 'models'): Promise<GithubObject[]>
   /**
-   * TODO: clean up:
+   * Clean up:
    *- .openapi-generator
    *- .gitignore
    *- .openapi-generator-ignore
    *- git_push.sh
    */
 
-  return model
+  return Promise.all([
+    ...redundantFiles.map((object) => fsPromises.unlink(`${model.outputPath}/${object}`)),
+    ...redundantDirectories.map((object) =>
+      fsPromises.rmdir(`${model.outputPath}/${object}`, { recursive: true }),
+    ),
+  ]).then((_) => model) // eslint-disable-line @typescript-eslint/no-unused-vars
+}
+
+function verifyObjectDefinition(definitions: Record<string, any>, key: string): boolean {
+  const definitionType = definitions[key].type
+
+  return (
+    (definitionType === 'object' &&
+      !exclusionExportingObjects.has(key) &&
+      /**
+       * The 'properties' keyword is used to define the object properties.
+       * Docs: https://swagger.io/docs/specification/data-models/data-types/#ctxM:~:text=The%20properties%20keyword%20is%20used%20to%20define%20the%20object%20properties
+       */
+      has(definitions[key], 'properties') &&
+      /**
+       * 'additionalProperties' is used to define a free form object.
+       * Docs: https://swagger.io/docs/specification/data-models/dictionaries/#free-form:~:text=Free%2DForm%20Objects
+       */
+
+      isEmpty(definitions[key].additionalProperties)) ||
+    /**
+     * Use to combine schemas
+     * Docs: https://swagger.io/docs/specification/data-models/oneof-anyof-allof-not/
+     */
+    has(definitions[key], 'oneOf') ||
+    has(definitions[key], 'allOf') ||
+    has(definitions[key], 'oneOf') ||
+    /**
+     *
+     */
+    (definitionType === 'string' && has(definitions[key], 'enum'))
+  )
 }
 
-function exportAPIModel(model: APIModel): APIModel {
-  // TODO: export models into src/index.ts and commit files
+async function exportAPIModel(model: APIModel): Promise<string> {
+  return SwaggerParser.parse(model.download_url).then(({ definitions }) => {
+    const exportings: string[] = []
 
-  return model
+    for (const key in definitions) {
+      if (has(definitions, key) && verifyObjectDefinition(definitions, key)) {
+        exportings.push(`${key} as ${upperFirst(camelCase(model.modelName))}${key}`)
+      }
+    }
+
+    return `export { ${exportings.join(', ')} } from './${model.modelName}'`
+  })
 }
 
 async function generateModels() {
-  const githubFolders = await fetchContentsByPath()
-  const githubFilePromises = githubFolders.map((folder) => fetchContentsByPath(folder.path))
+  const githubDirectories = await fetchContentsByPath()
+  const githubFilePromises = githubDirectories.map((directory) =>
+    fetchContentsByPath(directory.path),
+  )
 
   const githubFiles = await Promise.all(githubFilePromises)
 
-  return githubFiles
+  const apiModelGeneratorPromises = githubFiles
     .flat()
     .map(generateAPIModel)
     .map(executeGeneratorCLI)
-    .map(removeRedundantObjects)
-    .map(exportAPIModel)
+
+  await Promise.all<APIModel>(apiModelGeneratorPromises)
```

Comment:
Not clear why we need to mix `await` and `then`.

Can we use `await` for every call here?

Reply:
@moltar : Something like this?

```
  const apiModels = await Promise.all<APIModel>(apiModelGeneratorPromises)
  await Promise.all([
    ...apiModels.map(removeRedundantObjects),
    apiModels.map(generateExportStatement).then(writeStatementsToFile),
  ])
```

Reply:
Refactor `.then(writeStatementsToFile)` too :)